src/main/scala/com/evolution/scache/CacheFenced.scala

```diff
@@ -52,6 +52,9 @@ object CacheFenced {
           .productR { cache.put(key, value, release) }
       }
 
+      def modify[A](key: K, f: Option[V] => (A, Cache.Directive[F, V])): F[(A, Option[F[Unit]])] =
+        fence.flatMap(_ => cache.modify(key, f)) // TODO: this doesn't guarantee a race, though
```

Comment:
I think I didn't finish editing it, but basically the problem is that fence might close while modify is already running, so it might add a new releasable resource. Not sure if it's a big issue, though.